src/Monolog/Formatter/LogfmtFormatter.php

```diff
@@ -44,13 +45,15 @@ class LogfmtFormatter extends NormalizerFormatter
      * @param string|null $channelKey Key to use for the log channel name.
      * @param string|null $messageKey Key to use for the log message.
      * @param string|null $dateFormat The format of the timestamp: should be a format supported by DateTime::format
+     * @param string|null $lineSuffix The suffix to append at the end of the formatted line, defaults to a newline. (useful to set to null for syslog)
      */
     public function __construct(
         ?string $dateTimeKey = 'ts',
         ?string $levelKey = 'lvl',
         ?string $channelKey = 'chan',
         ?string $messageKey = 'msg',
-        ?string $dateFormat = DateTime::RFC3339
+        ?string $dateFormat = DateTime::RFC3339,
+        ?string $lineSuffix = "\n"
```

Comment:
Would you mind renaming `lineSuffix` to `formattedRecordTerminator`?

I just feel like since we're not necessarily outputting a "line" from `format` when this is set to something other than a newline, the slightly more verbose name might help make clear what it does.